tests/XRootD/scitokens.sh

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env bash
+
+function setup_scitokens() {
+	require_commands curl
+	if [ -z "$HOSTNAME" ]; then
+		HOSTNAME=`hostname`
+	fi
+
+	cat > "$PWD/scitokens/authdb" << EOF
+u * /issuer lr
+EOF
+	cat > "$PWD/scitokens/scitokens.cfg" <<EOF
+[Global]
+audience = https://$HOSTNAME:8095
+
+[Issuer test]
+issuer = https://$HOSTNAME:8095/issuer/one
+base_path = /protected
+EOF
+
+	# Setup the issuer and protected directory
+	ln -sf "$PWD/../issuer/export" "$PWD/scitokens/xrootd/issuer"
+	mkdir -p "$PWD/scitokens/xrootd/protected"
+	echo 'Hello, World' > "$PWD/scitokens/xrootd/protected/hello_world.txt"
+
+	# Override the scitoken cache location; otherwise, contents of prior test runs may be cached
+	export XDG_CACHE_HOME=$(mktemp -d "${PWD}/${NAME}/cache-XXXXXX")
+	mkdir -p "$XDG_CACHE_HOME/scitokens"
+
+	# Create a read-only token
+	$PWD/../scitokens/xrdscitokens-create-token ../issuer/issuer_pub_1.pem ../issuer/issuer_key_1.pem test_1 https://$HOSTNAME:8095/issuer/one storage.read:/ > "$PWD/scitokens/token"
+	if [ "$?" -ne 0 ]; then
```

Comment:
Is there a reason not to use [`assert`](https://github.com/xrootd/xrootd/blob/8a4ca755c57bd01b807a74a4aa3fc89b3f42116b/tests/XRootD/test.sh#L9-L11) command here, which was created for this? Also, the very long lines are not nice. Let's either break them or prepend `$PWD/../scitokens` to `PATH`, `pushd issuer` then run the command, it will be a lot more readable.

You could also `export ISSUER=https://$HOSTNAME:8095/issuer/one` and use that.

Reply:
Refactored quite a bit around here -- this is now wrapped in a function (which makes it far easier to see what's being tested).

`assert` couldn't be used because I need to redirect the output of `curl` and that's not compatible with the function.  I tried wrapping everything into an invocation of `/bin/sh` with a one-liner but it made things into an unreadable mess of quotes (and wasn't ultimately successful).  You're welcome to try to re-introduce it but I wasn't able to figure out an approach that made things better, not worse.